lib/domain.js

```diff
@@ -64,18 +82,37 @@ exports.create = exports.createDomain = function() {
 
 // the active domain is always the one that we're currently in.
 exports.active = null;
+Domain.prototype.members = undefined;
 
+Domain.prototype._enable = function() {
 
-inherits(Domain, EventEmitter);
+  const self = this;
+  this.asyncHook = async_hooks.createHook({
+    init(asyncId) {
 
-function Domain() {
-  EventEmitter.call(this);
+      if (process.domain === self) { // if this operation is created while in a domain, let's mark it
```

Comment:
You probably also want to set `resource.domain = process.domain` in this case

Reply:
Done, good point!